### PR TITLE
Issue #341

### PR DESCRIPTION
**Pull request by benoror**
Issue https://github.com/formly-js/angular-formly/issues/341

@kentcdodds please take this PR with caution as it can be completely wrong, most of all I would like to know your feedback, for my own benefit on learning formly's architecture.

### CODE REVIEW

File: src/directives/formly-field.js [hunk 1]
```diff
@@ -92,14 +92,16 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
...
 
-    function setFieldId() {
+    function setFieldIdAndName() {
       if (angular.isFunction(formlyConfig.extras.getFieldId)) {
         $scope.id = formlyConfig.extras.getFieldId($scope.options, $scope.model, $scope);
       } else {
         const formName = ($scope.form && $scope.form.$name) || $scope.formId;
         $scope.id = formlyUtil.getFieldId(formName, $scope.options, $scope.index);
       }
+      $scope.name = $scope.options.name || $scope.options.id;
```
Comment (kentcdodds) [inline src/directives/formly-field.js]: I think this line should be bumped down one line.

**Comment by kentcdodds** [comment]
This looks awesome! Just that one thing I mentioned, but I don't see any other issues. It looks great!

**Comment by benoror** [comment]
Cool! I have some doubts regarding the tests, you said `name` could be set both on the **scope** and the **field options object**. I'm not sure that both are correctly tested

File: src/directives/formly-field.js (hunk 1 above)
Comment (benoror) [inline src/directives/formly-field.js]: sorry for the noobish question, what's the cleaner way to amend this as a single commit? maybe [git revert](https://github.com/blog/2019-how-to-undo-almost-anything-with-git)?

Comment (kentcdodds) [inline src/directives/formly-field.js]: No worries! I'm no git expert either. But I think that a git revert may not be the best way... One second...

Comment (kentcdodds) [inline src/directives/formly-field.js]: I think that what you'll want to do is:

```
$ git reset --soft HEAD^
$ git commit -am 'Your message'
```

Comment (kentcdodds) [inline src/directives/formly-field.js]: You'll have to do a force push as well, so beware... `git push -f`

**Comment by benoror** [comment]
Oops it seems building failed because of `sinion.js` issues

File: src/directives/formly-field.js [hunk 2]
```diff
@@ -100,6 +100,8 @@ function formlyField($http, $q, $compile, $templateCache, $interpolate, formlyCo
         $scope.id = formlyUtil.getFieldId(formName, $scope.options, $scope.index);
       }
       $scope.options.id = $scope.id;
+      $scope.name = $scope.options.name || $scope.options.id;
```
Comment (benoror) [inline src/directives/formly-field.js]: done!

**Comment by kentcdodds** [comment]
Looks like the build is breaking because `sinon` recently stopped publishing a `pkg` file and `karma-sinon` attempts to add the `sinon` file from `pkg` to the `files` in the karma config. I'll look into that problem, then this will get merged. Thanks!

**Comment by kentcdodds** [comment]
Looks like a lot of people are aware of this issue: https://github.com/cjohansen/Sinon.JS/issues/761

**Comment by benoror** [comment]
Ok, so the tests look good as well? https://github.com/formly-js/angular-formly/pull/346#issuecomment-110844936

**Comment by kentcdodds** [comment]
They must. If you were able to commit this code, then they passed on your machine :-) I'm pushing up a fix, then I'll merge your stuff.

**Pull request merged**